### PR TITLE
os: fix posix_spawn exception

### PR DESCRIPTION
**Pull request by deantvv**

### CODE REVIEW

File: vm/src/stdlib/posix.rs [hunk 1]
```diff
@@ -1278,7 +1278,7 @@ pub mod module {
         fn spawn(self, spawnp: bool, vm: &VirtualMachine) -> PyResult<libc::pid_t> {
             use crate::TryFromBorrowedObject;
 
-            let path = CString::new(self.path.into_bytes())
+            let path = CString::new(self.path.clone().into_bytes())
```
Comment (youknowone) [inline vm/src/stdlib/posix.rs:1281]:
```suggestion
            let path = CString::new(self.path.as_bytes())
```
is this working?

**Review by youknowone — APPROVED** [review]

File: vm/src/stdlib/posix.rs (hunk 1 above)
Comment (deantvv) [reply vm/src/stdlib/posix.rs:1281]: No, we don't have `as_bytes` for `PyPathLike` but IMO it makes sense to implement it.

Comment (youknowone) [reply vm/src/stdlib/posix.rs:1281]: Probably not. `Path` doesn't look like to be borrowable as bytes.

Comment (youknowone) [reply vm/src/stdlib/posix.rs:1281]: But I found another utility to do this job
```suggestion
            let path = self.path.into_cstring()?;
```

Comment (deantvv) [reply vm/src/stdlib/posix.rs]: We need to clone for this to work.

**Pull request merged**